File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let answer = 0;
    let stack = [];
    let isCorrect = true;
  
    if (s.length % 2 === 1) return 0;

    for (let i=0; i<s.length; i++) {       
       let str = s.slice(i) + s.slice(0,i);
       isCorrect = true;
        for (let n of str) {
            if(n === "[" || n === "{" || n === "(" ){
                stack.push(n);
            } else {
                let opening = stack.pop();
                if (opening === "(" && n === ")") continue;
                if (opening === "{" && n === "}") continue;
                if (opening === "[" && n === "]") continue;
              	isCorrect = false;
                break;
            };
        };
        if (isCorrect) answer++;
    };
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    
    if len(s) % 2 != 0:
        return 0
    bracket_map = {')': '(', '}': '{', ']': '['}
    
    for i in range(len(s)):
        rotated_s = s[i:] + s[:i]
        stack = []
        is_current_rotation_correct = True
        
        for char in rotated_s:
            if char in ['(', '{', '[']:
                stack.append(char)
            elif char in [')', '}', ']']:
                if not stack:
                    is_current_rotation_correct = False
                    break
                opening_bracket = stack.pop()
                if opening_bracket != bracket_map[char]:
                    is_current_rotation_correct = False
                    break
        if is_current_rotation_correct and not stack:
            answer += 1
    return answer